plugin/evm/gossip_test.go

```diff
@@ -24,3 +36,67 @@ func TestGossipEthTxMarshaller(t *testing.T) {
 	require.NoError(err)
 	require.Equal(want.GossipID(), got.GossipID())
 }
+
+func TestGossipSubscribe(t *testing.T) {
+	require := require.New(t)
+	key, err := crypto.GenerateKey()
+	require.NoError(err)
+	addr := crypto.PubkeyToAddress(key.PublicKey)
+
+	require.NoError(err)
+	txPool := setupPoolWithConfig(t, params.TestChainConfig, addr)
+	defer txPool.Stop()
+	txPool.SetGasPrice(common.Big1)
+	txPool.SetMinFee(common.Big0)
+
+	gossipTxPool, err := NewGossipEthTxPool(txPool)
+	require.NoError(err)
+
+	// use a custom bloom filter to test the bloom filter reset
+	gossipTxPool.bloom, err = gossip.NewBloomFilter(1, 0.01, 0.0000000000000001) // maxCount =1
+	require.NoError(err)
+	ctx, cancel := context.WithCancel(context.TODO())
+	go func() {
+		gossipTxPool.Subscribe(ctx)
+	}()
+
+	// create first eth txes
+	ethTxs := getValidEthTxs(key, 10, big.NewInt(226*params.GWei))
+
+	// Notify VM about first eth txs batch
+	batch1 := ethTxs[:5]
+	errs := txPool.AddRemotesSync(batch1)
+	for _, err := range errs {
+		require.NoError(err, "failed adding subnet-evm tx to remote mempool")
+	}
+
+	// Notify VM about second eth txs batch
+	batch2 := ethTxs[5:]
+	errs = txPool.AddRemotesSync(batch2)
+	for _, err := range errs {
+		require.NoError(err, "failed adding subnet-evm tx to remote mempool")
+	}
+	time.Sleep(50 * time.Millisecond)
+	cancel()
```

Comment:
is there a way to avoid this pattern, since it may lead to flakes?

Reply:
I intended to use VM, but ethTxPool was exported/not possible to tweak maxCount with bloom parameters. I just increased it to 1 sec to be sure, if this flakes we can try to change the actual ethTxPool structure.